src/measure_theory/vitali_caratheodory.lean

```diff
@@ -0,0 +1,519 @@
+    simp only [simple_func.coe_add, ennreal.coe_add, pi.add_apply],
+    rw [lintegral_add f₁.measurable.coe_nnreal_ennreal f₂.measurable.coe_nnreal_ennreal,
+        lintegral_add g₁cont.measurable.coe_nnreal_ennreal g₂cont.measurable.coe_nnreal_ennreal],
+    convert add_le_add g₁int g₂int using 1,
+    conv_lhs { rw ← ennreal.add_halves ε },
+    abel }
+end
+
+open simple_func (eapprox_diff tsum_eapprox_diff)
+
+/-- Given a measurable function `f` with values in `ℝ≥0`, there exists a lower semicontinuous
+function `g ≥ f` with integral arbitrarily close to that of `f`. Formulation in terms of
+`lintegral`.
+Auxiliary lemma for Vitali-Carathéodory theorem `exists_lt_lower_semicontinuous_integral_lt`. -/
+lemma exists_le_lower_semicontinuous_lintegral_ge
+  (f : α → ℝ≥0∞) (hf : measurable f) {ε : ℝ≥0∞} (εpos : 0 < ε) :
+  ∃ g : α → ℝ≥0∞, (∀ x, f x ≤ g x) ∧ lower_semicontinuous g ∧ (∫⁻ x, g x ∂μ ≤ ∫⁻ x, f x ∂μ + ε) :=
+begin
+  rcases ennreal.exists_pos_sum_of_encodable' εpos ℕ with ⟨δ, δpos, hδ⟩,
+  have : ∀ n, ∃ g : α → ℝ≥0, (∀ x, simple_func.eapprox_diff f n x ≤ g x) ∧ lower_semicontinuous g ∧
+    (∫⁻ x, g x ∂μ ≤ ∫⁻ x, simple_func.eapprox_diff f n x ∂μ + δ n) :=
+  λ n, simple_func.exists_le_lower_semicontinuous_lintegral_ge μ
+    (simple_func.eapprox_diff f n) (δpos n),
+  choose g f_le_g gcont hg using this,
+  refine ⟨λ x, (∑' n, g n x), λ x, _, _, _⟩,
+  { rw ← tsum_eapprox_diff f hf,
+    exact ennreal.tsum_le_tsum (λ n, ennreal.coe_le_coe.2 (f_le_g n x)) },
+  { apply lower_semicontinuous_tsum (λ n, _),
+    exact ennreal.continuous_coe.comp_lower_semicontinuous (gcont n)
+      (λ x y hxy, ennreal.coe_le_coe.2 hxy) },
+  { calc ∫⁻ x, ∑' (n : ℕ), g n x ∂μ
+    = ∑' n, ∫⁻ x, g n x ∂μ :
+      by rw lintegral_tsum (λ n, (gcont n).measurable.coe_nnreal_ennreal)
+    ... ≤ ∑' n, (∫⁻ x, eapprox_diff f n x ∂μ + δ n) : ennreal.tsum_le_tsum hg
+    ... = ∑' n, (∫⁻ x, eapprox_diff f n x ∂μ) + ∑' n, δ n : ennreal.tsum_add
+    ... ≤ ∫⁻ (x : α), f x ∂μ + ε :
+      begin
+        refine add_le_add _ hδ.le,
+        rw [← lintegral_tsum],
+        { simp_rw [tsum_eapprox_diff f hf, le_refl] },
+        { assume n, exact (simple_func.measurable _).coe_nnreal_ennreal }
+      end }
+end
+
+/-- Given a measurable function `f` with values in `ℝ≥0` in a sigma-finite space, there exists a
+lower semicontinuous function `g > f` with integral arbitrarily close to that of `f`.
+Formulation in terms of `lintegral`.
+Auxiliary lemma for Vitali-Carathéodory theorem `exists_lt_lower_semicontinuous_integral_lt`. -/
+lemma exists_lt_lower_semicontinuous_lintegral_ge [sigma_finite μ]
+  (f : α → ℝ≥0) (fmeas : measurable f) {ε : ℝ≥0} (εpos : 0 < ε) :
+  ∃ g : α → ℝ≥0∞, (∀ x, (f x : ℝ≥0∞) < g x) ∧ lower_semicontinuous g ∧
+    (∫⁻ x, g x ∂μ ≤ ∫⁻ x, f x ∂μ + ε) :=
+begin
+  rcases exists_integrable_pos_of_sigma_finite μ (nnreal.half_pos εpos) with ⟨w, wpos, wmeas, wint⟩,
+  let f' := λ x, ((f x + w x : ℝ≥0) : ℝ≥0∞),
```

Comment:
If I understand correctly, the argument here to make the pointwise inequality strict is not part of the standard Vitali-Carathéodory theorem.  And it has the cost of a sigma-finiteness assumption in every theorem that follows.

Are you taking the extra step because it's needed for your follow-up PR on the FTC?  If so, could you please add a comment about this?